src/gsi/gsi_rfv3io_mod.f90

```diff
@@ -3190,5 +3060,102 @@ subroutine convert_cvpnx_to_nx(qnx_arr,cvpnr,cvpnr_pvalue,cloud_nt_updt,q_arr,qr
+   integer(i_kind):: ivar,jvar,istatus
+   src_nc3d=bundi%n3d
+   src_nc2d=bundi%n2d
+   target_nc3d=bundo%n3d
+   target_nc2d=bundo%n2d
+   allocate(src_name_vars3d(src_nc3d),src_name_vars2d(src_nc2d))
+   allocate(target_name_vars3d(target_nc3d),target_name_vars2d(target_nc2d))
+   call gsi_bundleinquire(bundi,'shortnames::3d',src_name_vars3d,istatus)
+   call gsi_bundleinquire(bundi,'shortnames::2d',src_name_vars2d,istatus)
+   call gsi_bundleinquire(bundo,'shortnames::3d',target_name_vars3d,istatus)
+   call gsi_bundleinquire(bundo,'shortnames::2d',target_name_vars2d,istatus)
+   do ivar=1,src_nc3d
+      varname=trim(src_name_vars3d(ivar))
+      do jvar=1,target_nc3d
+        if(index(target_name_vars3d(jvar),varname).gt.0)  then
+        call GSI_BundleGetPointer (bundi,varname,pvar3d,istatus)
+        call gsi_bundleputvar (bundo,varname,pvar3d,istatus)
+!         write(6,*)varname ,' is copied from bundi to bundo in  gsi_copy_bundle'
+        exit
+        endif
+       enddo
+    enddo
+    do ivar=1,src_nc2d
+       varname=trim(src_name_vars2d(ivar))
+       do jvar=1,target_nc2d
+         if(index(target_name_vars2d(jvar),varname).gt.0)  then
+         call GSI_BundleGetPointer (bundi,varname,pvar2d,istatus)
+         call gsi_bundleputvar (bundo,varname,pvar2d,istatus)
+!         write(6,*)'varname 2d',trim(varname),' is copied from bundi to bundo in  gsi_copy_bundle'
+         exit
+         endif
+        enddo
+    enddo
+    deallocate(src_name_vars3d,src_name_vars2d)
+    deallocate(target_name_vars3d,target_name_vars2d)
+    return
+    end subroutine gsi_copy_bundle
+    subroutine getfv3lamfilevname(vgsinamein,fv3filenamegref,filenameout,vname)
+    type (type_fv3regfilenameg),intent (in) :: fv3filenamegref
+    character(len=*):: vgsinamein
+    character(len=*),intent(out):: vname
+    character(len=*),intent(out):: filenameout
+    if (ifindstrloc(vgsiname,vgsinamein)<= 0) then
+      write(6,*)'the name ',vgsinamein ,'cannot be treated correctly in getfv3lamfilevname,stop'
+      call stop2(333)
+    endif
+    if(ifindstrloc(vardynvars,vgsinamein)> 0)  then 
+        filenameout=fv3filenamegref%dynvars
+    else if(ifindstrloc(vartracers,vgsinamein)> 0 )  then 
+        filenameout=fv3filenamegref%tracers
+    else
+        write(6,*)'the filename corresponding to var ',trim(vgsinamein),' is not found, stop ' 
+        call stop2(333)
+    endif
+    vname=varfv3name(ifindstrloc(vgsiname,vgsinamein))
+    if(trim(vname)=="T".and. fv3sar_bg_opt==1) then
+       vname="t"
+    endif 
+    
+    return
+    end subroutine getfv3lamfilevname
+    function ifindstrloc(str_array,strin)
+    integer(i_kind) ifindstrloc
+    character(len=max_varname_length),dimension(:) :: str_array
+    character(len=*) :: strin
+    integer(i_kind) i
+    ifindstrloc=0
+    do i=1,size(str_array)
+      if(trim(str_array(i)) == trim(strin)) then 
+      ifindstrloc=i
+      exit
+      endif
```

Comment:
Please add proper indentation to lines 3153 through 3156.

Reply:
Mike, So many thanks for your details identification of those formatting issues.  This PR contained a significant refactoring of previous codes.  While I tried to minimalize the changes by copying/pasting ,moving around different pieces from the previous version, I should have paid more attention to the formats required by GSI standard.  I have made changes following your comments (while I didn't reply to avoid too many messages sent out.  I will update this modified PR following your instructions and let you know when it is done